### DISCUSSION TITLE
Chapter title pages/changing the page layout for single pages

### DISCUSSION THREAD

**Discussion by fruchti** [typst/typst#1082] [Q&A]
I’m currently trying to modify the page style for every page with a chapter heading on it. This involves header and footer texts as well as different page margins from the rest. Basically, I’m looking for something more or less like `\thispagestyle` in LaTeX.

The first step, adding a page break before each chapter, is not an issue:

```typst
#show heading.where(level: 1): it => [
    #pagebreak(weak: true)
    #it
]
```

However, I haven’t found a way to `set page()` the way I want. Things I’ve tried so far:

- Using `show/set` (e.g. `#show heading.where(level: 1): set page(..title_page_options)`) effects a page break immediately after the heading. I assume since the page style is only set for the heading’s block, typst begins a new page to assure that the next block has the old page style again.
- Defining a function (used instead of `= Chapter Title`) which includes a `#set page` has the same result.
- I might be able to build a selector out of `before()` and `after()` which gets me pages with a chapter heading on them. With this, I could adjust the header/footer text like [the example in the docs](https://typst.app/docs/reference/meta/query/). Since `locate()` returns a `content`, I cannot use this to adjust the margins. I also cannot adjust all values at once, since I didn’t find a way to pass a dictionary through `locate()`.
- With `#show page: p => locate(loc => p)` I got no output at all any more, so I didn’t try any further based on that.
- I tried using `#set… if` together with `locate()` by returning `content` from `locate()` which either has or hasn’t a specific field, depending on whether I am currently on a page with a chapter heading or not. However, even with
  ```
  #set page(..title_page_options) if locate(loc => box(width: 1pt)).has("width")
  ```
  the condition never holds and the style doesn’t get applied (it looks like the `locate()` isn’t even evaluated actually).

**Comment by Joelius300** [comment]
Interestingly, the first snippet already gives me an error.

![image](https://github.com/typst/typst/assets/52202086/fe9573da-554b-44f7-9019-7c9c78071f05)

**Reply by laurmaedje** [reply]
This happens if you are in a `columns` container. It's a long-standing limitation. The easiest workaround is to use a `colbreak` instead.
